backend/controllers/vehicleController.js

```diff
@@ -64,15 +75,25 @@ exports.updateVehicle = asyncHandler(async (req, res, next) => {
 exports.createVehicle = asyncHandler(async (req, res, next) => {
   const reqBody = req.body;
 
+  // grab the dealership_ID from the body and verify that the dealership exists
+  const dealership = await Dealership.findById(req.body.dealership);
+
+  // no dealership found
+  if (!dealership) {
+      return next(
+          new ErrorResponse('dealership not found. Cannot create a vehicle without a valid dealership.', 400)
+      );
+  }
+
   // try to create the new vehicle
   const newVehicle = await Vehicle.create(reqBody);
 
   // return an error response if no vehicle was created
-  if (!newVehicle) {
-    return next(
-      new ErrorResponse("Vehicle was not created. Please validate your input.")
-    );
-  }
+  // if (!newVehicle) {
+  //   return next(
+  //     new ErrorResponse("Vehicle was not created. Please validate your input." , 400)
+  //   );
+  // }
```

Comment:
We should keep this check in since it catches any error scenarios that we could have forgotten.

Reply:
After further review, this code seems to be unnecessary and it was deleted because we already have a check in the middleware, and i have added the check in case the dealership_id is invalid.

Reply:
Yes it seems like an error is thrown instead of returning an empty vehicle and then caught in the middleware. This code would also be unreachable indeed.